Task2/aga/TicTacToe/TicTacToe/TheGame.xaml.cs

```diff
@@ -0,0 +1,242 @@
+        }
+
+        private void B2_Click(object sender, RoutedEventArgs e)
+        {
+            Button b = (Button)sender;
+            int turn = click.GetTurn();
+            int param = 4;
+
+            displayButton(turn, b, param);
+        }
+
+        private void B3_Click(object sender, RoutedEventArgs e)
+        {
+            Button b = (Button)sender;
+            int turn = click.GetTurn();
+            int param = 5;
+
+            displayButton(turn, b, param);
+        }
+
+        private void C1_Click(object sender, RoutedEventArgs e)
+        {
+            Button b = (Button)sender;
+            int turn = click.GetTurn();
+            int param = 6;
+
+            displayButton(turn, b, param);
+        }
+
+        private void C2_Click(object sender, RoutedEventArgs e)
+        {
+            Button b = (Button)sender;
+            int turn = click.GetTurn();
+            int param = 7;
+
+            displayButton(turn, b, param);
+        }
+        private void C3_Click(object sender, RoutedEventArgs e)
+        {
+            Button b = (Button)sender;
+            int turn = click.GetTurn();
+            int param = 8;
+
+            displayButton(turn, b, param);
+        }
+
+        private void displayTurn(int t)
+        {
+            Label l = diplayTurn;
+            l.Content = t;
+        }
+
+        private void displayMessageBox(int param, int turn)
+        {
+            Label l = messageBox;
+            
+            if (wasWinned())
+                { l.Content = "The winner is "+ winner; }
+            else if (turn == 9)
+                {
+                    l.Content = "It's a draw!";
+                    tryAgain.Visibility = Visibility.Visible;
+                }
+            else if (wasClicked(param))
+            { l.Content = "Choose different field"; }
+            else
+                {
+                    if (turn % 2 != 0)
+                    {
+                        l.Content = "Player O turn";
+                    }
+                    else
+                    {
+                        l.Content = "Player X turn";
+                    }
+                }
+        }
+
+        private void displayButton(int turn, Button b, int param)
+        {
+            
+            if (!wasClicked(param) && !wasWinned())
+
+            {   
+                if (turn % 2 != 0)
+                {
+                    b.Content = "X";
+                }
+                else
+                {
+                    b.Content = "O";
+                }
+                displayMessageBox(param, turn);
+                clickedButtons[param] = true;
+                click.TurnCounter();
+                displayTurn(turn);
+            }
+            else
+                { displayMessageBox(param, turn); }
+            
+        }
+
+        private bool wasClicked(int param)
+        {
+            return clickedButtons[param];           
+        }
+
+        private bool wasWinned()
```

Comment:
There's a lot of code duplication here. Think of how you could minimize it. If you have multiple fields on a board, maybe you could use a collection and run some algorithm on that collection instead of treating each field as a separate thing?